velox/functions/sparksql/specialforms/SparkCastExpr.cpp

```diff
@@ -32,8 +32,8 @@ exec::ExprPtr SparkCastCallToSpecialForm::constructSpecialForm(
       type,
       std::move(compiledChildren[0]),
       trackCpuUsage,
-      false,
-      std::make_shared<SparkCastHooks>(config));
+      true,
+      std::make_shared<SparkCastHooks>(config, false));
```

Comment:
Why we are passing true to the SparkCastExpr as the isTryCast?

Reply:
Spark’s cast function (with ANSI mode disabled) and its try_cast function must both return NULL when given invalid input, which is why isTryCast needs to be set to true. The difference between them is how they treat overflow: the regular cast does not consider overflow as invalid input, whereas try_cast does treat overflow as illegal input. This behavior is determined by the specific cast policy in use.

Reply:
It appears to be a bit conflicting as we are setting `isTryCast` in CastExpr as true, but in SparkCastHooks as false. The reason could be that Spark's try_cast and Velox's TryCast have distinct semantics.

> The difference between them is how they treat overflow: the regular cast does not consider overflow as invalid input, whereas try_cast does treat overflow as illegal input. This behavior is determined by the specific cast policy in use.

Would it make sense to rename the Spark try-cast parameter as allowOverflow?

Reply:
I renamed the isTryCast parameter in SparkCastHooks to allowOverflow. Since under the current semantics isTryCast is always true, I removed this parameter from the SparkCastExpr constructor and added a comment explaining the change. I thought the confusion stems from the semantics of Spark’s cast (ANSI off) functions, and there may not be a cleaner mapping from spark cast to velox cast.